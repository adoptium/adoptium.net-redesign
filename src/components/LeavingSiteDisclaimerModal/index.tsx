import React, { useState } from "react";
import { Trans } from 'gatsby-plugin-react-i18next';

const LeavingSiteDisclaimerModal = () => {

  const [message, setMessage] = useState('');
  const [location, setLocation] = useState('');

  if (typeof window !== `undefined`) {
    window.addEventListener('show.bs.modal', (event: Event) => {
      let button = event.relatedTarget;
      setMessage(button.getAttribute('data-bs-message'))
      setLocation(button.getAttribute('data-bs-location'))
    });
  }

  return (
    <>
      <div className="modal fade" id="leavingSiteDisclaimerModal" tabIndex={-1} aria-labelledby="leavingSiteDisclaimerLabel" aria-hidden="true" style={{zIndex: '10000'}}>
        <div className="modal-dialog modal-lg">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title" id="leavingSiteDisclaimerLabel">
                <Trans i18nKey='leaving.site.disclaimer.modal.title' defaults='Disclaimer - You are going to leave our website' />
              </h5>
              <button type="button" className="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
            </div>
            <div className="modal-body text-start">
              <p>
                {message}           
              </p>
            </div>
            <div className="modal-footer">
              <button 
                type="button"
                className="btn btn-primary"
                onClick={(event: React.MouseEvent<HTMLButtonElement>): void => {
                  event.preventDefault();
                  window.location.assign(location);
                }}
              >
                <Trans i18nKey='leaving.site.disclaimer.modal.continue' defaults='Continue' />
              </button>
              <button type="button" className="btn btn-secondary" data-bs-dismiss="modal">
                <Trans i18nKey='leaving.site.disclaimer.modal.close' defaults='Close' />
              </button>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default LeavingSiteDisclaimerModal;
