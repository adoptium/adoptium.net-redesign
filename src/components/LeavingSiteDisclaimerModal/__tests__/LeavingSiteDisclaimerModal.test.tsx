import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest'
import LeavingSiteDisclaimerModal from '..';

describe('LeavingSiteDisclaimerModal component', () => {
  it('LeavingSiteDisclaimerModal renders correctly', () => {
    const { container } = render(
      <LeavingSiteDisclaimerModal />
    );
    expect(container).toMatchSnapshot();
  });

  it('LeavingSiteDisclaimerModal continue correctly', async () => {
    const { container } = render(
      <LeavingSiteDisclaimerModal />
    );

    const cancelButton = container.querySelector('button[class="btn btn-primary"]');

    await userEvent.click(cancelButton).then(async() => {
      expect(container).toMatchSnapshot();
    });
  });
});
