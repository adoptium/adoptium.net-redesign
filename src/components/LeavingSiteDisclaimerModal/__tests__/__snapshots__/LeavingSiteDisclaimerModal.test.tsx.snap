// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LeavingSiteDisclaimerModal component > LeavingSiteDisclaimerModal continue correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="leavingSiteDisclaimerLabel"
    class="modal fade"
    id="leavingSiteDisclaimerModal"
    style="z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="leavingSiteDisclaimerLabel"
          >
            Text
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body text-start"
        >
          <p />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Text
          </button>
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Text
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LeavingSiteDisclaimerModal component > LeavingSiteDisclaimerModal renders correctly 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="leavingSiteDisclaimerLabel"
    class="modal fade"
    id="leavingSiteDisclaimerModal"
    style="z-index: 10000;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-lg"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title"
            id="leavingSiteDisclaimerLabel"
          >
            Text
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            type="button"
          />
        </div>
        <div
          class="modal-body text-start"
        >
          <p />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Text
          </button>
          <button
            class="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            Text
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
